apollo-federation/src/merger/merge_links.rs

```diff
@@ -43,6 +45,33 @@ struct CoreDirectiveInSupergraph {
     composition_spec: DirectiveCompositionSpecification,
 }
 
+fn get_directive_source_and_import(
```

Comment:
This duplicates the logic of `LinksMetadata::source_link_of_directive`. Maybe this could be simplified using it?